pkgs/development/libraries/sword/default.nix

```diff
@@ -1,13 +1,14 @@
 { stdenv, fetchurl, pkgconfig, icu, clucene_core, curl }:
 
+with stdenv.lib;
```

Comment:
i see no reason to move the statement here, which can cause confusion later on.

Reply:
During the testing phases, I always use some functions from `stdenv.lib`, from string substitution to `fakeSha256`, and I don't see a reason why it can't be put in a broader context.

Reply:
It's just not that common, unless you have a good reason to do that... If you insist on using `with stdenv.lib`, then why not use `mkDerivation`? These are unnecessary changes for a mere update.

Reply:
the 'with' prevents from running some checks and thus can hide errors. It s very conveninent but better let it close to a minimal scope: I am not bikeshedding.

Reply:
>  prevents from running some checks 

How, exactly?

If it is a problem, then I will abolish `with` at all.

Reply:
for instance neomake uses nix-instantiate and when using `with`, it can't know if a variable is undefined or imported via the `with`, hence it may not trigger errors for undefined variables. I agree `with` is useful but it's best to limit the scope it applies to.

Reply:
It is a good argument, indeed. I will revise my expressions in order to 'fix' it.